src/encoded/commands/deploy.py

```diff
@@ -200,37 +204,80 @@ def create_ec2_instances(client, image_id, count, instance_type, security_groups
     )
     return reservations
 
+
 def tag_ec2_instance(instance, name, branch, commit, username, elasticsearch, cluster_name):
-    tags=[
+    tags = [
         {'Key': 'Name', 'Value': name},
         {'Key': 'branch', 'Value': branch},
         {'Key': 'commit', 'Value': commit},
         {'Key': 'started_by', 'Value': username},
     ]
     if elasticsearch == 'yes':
         tags.append({'Key': 'elasticsearch', 'Value': elasticsearch})
-    if cluster_name != None:
+    if cluster_name is not None:
         tags.append({'Key': 'ec_cluster_name', 'Value': cluster_name})
     instance.create_tags(Tags=tags)
     return instance
 
+
 def tag_spot_instance(instance, name, branch, commit, username, elasticsearch, client, cluster_name):
-    tags=[
+    tags = [
         {'Key': 'Name', 'Value': name},
         {'Key': 'branch', 'Value': branch},
         {'Key': 'commit', 'Value': commit},
         {'Key': 'started_by', 'Value': username},
     ]
     if elasticsearch == 'yes':
         tags.append({'Key': 'elasticsearch', 'Value': elasticsearch})
-    if not cluster_name == None:
+    if cluster_name is not None:
         tags.append({'Key': 'ec_cluster_name', 'Value': cluster_name})
     instance_id = client.create_tags(Resources=[get_instance_id(instance, client)], Tags=tags)
     return instance_id
 
-def run(wale_s3_prefix, image_id, instance_type, elasticsearch, spot_instance, spot_price, cluster_size, cluster_name, check_price,
-        branch=None, name=None, role='demo', profile_name=None, teardown_cluster=None):
-    
+
+def read_ssh_key():
+    home = expanduser("~")
+    ssh_key_path = home + '/' + '.ssh/id_rsa.pub'
+    ssh_keygen_args = ['ssh-keygen', '-l', '-f', ssh_key_path]
+    fingerprint = subprocess.check_output(
+        ssh_keygen_args
+    ).decode('utf-8').strip()
+    if fingerprint:
+        with open(ssh_key_path, 'r') as f:
+            ssh_pub_key = f.readline().strip()
+            return ssh_pub_key
+
+
+def get_user_data(commit, config_file, data_insert, profile_name):
+    cmd_list = ['git', 'show', commit + config_file]
+    config_template = subprocess.check_output(cmd_list).decode('utf-8')
+    ssh_pub_key = read_ssh_key()
+    if not ssh_pub_key:
+        print(
+            "WARNING: User is not authorized with ssh access to "
+            "new instance because they have no ssh key"
+        )
+    data_insert['LOCAL_SSH_KEY'] = ssh_pub_key
+    # aws s3 authorized_keys folder
+    auth_base = 's3://encoded-conf-prod/ssh-keys'
+    auth_type = 'prod'
```

Comment:
It would be safer if default value for auth_type was 'dev', and flipped to 'prod' only when specified otherwise. This can be achieved by setting default to 'dev' for '--profile-name' at argparse level. Although we don't follow the convention of always setting default argument values, in this case it is a neat opportunity.

Reply:
I thought about this too, but 'prod' was chosen in the end for the auth_type default because it is the most limiting.  This way ssh access defaults to limited admin keys instead of all keys.              

I have a refactor deploy script branch on the heals of this one.  Maybe we should revisit this in there?  I'm already rearranging the args in there so it's a better place to fix the defaults.